

<img src="./logo_UTN.svg" align="right" width="150" /> 
 Medidas Electrónica I

# Ejercicios de Osciloscopia y frecuencimetro
#### Israel Pavelek 

Prof: Ing. Perdomo Juan

Ayudante: Ing. Franco Zaccra


Se realiza la medición señal senoidal rectificada con el osciloscopio que se detallan a
continuación.

Osciloscopio Digital:  

* Ancho de Banda: 100MHz
* Impedancia de entrada:
* 1MΩ // 15pF FEV: 1V/div
* Error: ± (2% + 0,1div + 1mV)
* Base de tiempo: 2ms/div.
* Error: ±(0,5% + 0,1div)
* Punta de Prueba: X10
* Error: ±0,5%

Frecuencímetro:

* Base de tiempo: 10MHz
* Estabilidad: 5ppm
* Envejecimiento: 0,1ppm/mes
* (Considerar uso luego de 1 año de calibrado)
* Relación señal a ruido: 60dB
* Punto de disparo: punto medio de la señal

<img src="./Ej1.png" align="center" width="400" /> 

* a) Determinar el valor eficaz de la señal y su incertidumbre
* b) Determine la frecuencia de la señal y su incertidumbre utilizando el osciloscopio
* c) Determine la frecuencia de la señal y su incertidumbre utilizando el frecuencímetro en medición de periodo promediados X10. Indicación del contador: 519582
* d) Determine la frecuencia de la señal y su incertidumbre utilizando el frecuencímetro en medición de frecuencia con gate 10s, tener en cuenta la indicación del punto c para determinar la indicación de este punto.

## Calculo del valor eficaz

Para la medición de tension vemos que la señal excursiona por 6 divisiones del osciloscopio por lo que tenemos:

$Vpp=6 Div \cdot \frac{1V}{Div}= 6V$

$\Delta Vpp=(\frac{2 \cdot 6V}{100} + 0,1div \cdot \frac{1V}{div} + 1mV)= 221mv $

$\mu_R(Vpp)= \frac{\Delta Vpp}{\sqrt{3} \cdot 6V}= 0,0212657 $

$\mu_R(FP)= \frac{0,5}{100 \cdot \sqrt{3}}= 0,0028867 $

$V_{RMS}=\frac{Vpp}{\sqrt{2}} \cdot 10 =42,4264V $

$\mu_R (V_{RMS})=\sqrt{\mu_R^2(Vpp) + \mu_R^2(FP)} =0,021460766$

$\mu (V_{RMS}) = V_{RMS} \cdot \mu_R (V_{RMS}) = 0,12876460174 V $

$U(V_{RMS})= K \cdot \mu(V_{RMS})= 0,257529203481V$

$V_{RMS}=(42,42±0,26) V$

## Calculo de la frecuencia con el osciloscopio

$T=2,6 Div \cdot \frac{2ms}{Div}= 5,2ms$

$\Delta T=(\frac{0,5 \cdot 5,2ms}{100} + 0,1div \cdot \frac{2ms}{div})= 0,226ms $

$\mu_R(T)=\frac{\Delta T}{\sqrt{3}\cdot 5,2ms}=0,0250925309302$

$f=\frac{1}{T}=192,307692308Hz$

$\mu_R(T)=\mu_R(f)$

$\mu(f)=\mu_R(f) \cdot 192,307692308Hz=4,82548671916Hz$

$U(F)=K \cdot \mu(f)=9,65097343832 Hz$

$f=(192,3±9,7)Hz$

## Calculo de la frecuencia con frecuencimetro  en modo medicion de periodo promediado x 10

Estabilidad de la base de tiempo:

$\frac{\Delta f_{BT}}{f_{BT}}=5ppm+0,1\frac{ppm}{mes} \cdot 12 meses =6,2ppm$

Incertidumbre de cuantización:

$\frac{\Delta N}{N}=\frac{1}{519582}=1,924624ppm$

Error debido al disparo:

$V(t)=V_SPsin(\omega t)$ 

$\Delta V=V_{NP}$

$SNR=\frac{V_{SP}}{V_{NP}}$

$\Delta_T=\frac{\Delta V}{[\frac{dV}{dt}]_{t=t_{TRG}}}$

Sabemos que el disparo se produce cuando la señal está en $ \frac{V_{SP}}{2} $

$\frac{V_{SP}}{2}=V_{SP}sin(\omega t)$

$arcsin(\frac{1}{2})=\omega t$

$\frac{\pi}{\omega 6}=t$

$\Delta_T=\frac{V_{NP}}{[V_{SP}\cdot \omega \cdot cos(\omega 
t)]_{t=\frac{\pi}{6\omega}}}$

$\Delta_T=\frac{V_{NP}}{V_{SP}\cdot \omega \cdot \frac{\sqrt{3}}{2}}$

$\frac{\Delta_T}{T}=\frac{V_{NP}}{V_{SP}\sqrt3 \pi}$

$e_D=2\cdot \frac{\Delta_T}{T}$

$e_D=\frac{2}{\sqrt3 \pi SNR}$


Reemplazando:

$e_D=\frac{2}{\sqrt3 \pi 1000}=0,36755 \cdot 10^{-3}$

$\frac{\Delta_T}{T}=±(\frac{\Delta f_{BT}}{f_{BT}}+\frac{\Delta N}{N}+\frac{e_D}{M})= 375,6748 ppm$

M=1 (promediado por 10)

$\mu_R(T)=\frac{\frac{\Delta_T}{T}}{sqrt3}=216,8959 ppm$

$\mu_R(T)=\mu_R(f)$

$T=\frac{N}{F_{BT} \cdot 10^1}=5,19582 ms$

$f=\frac1T= 192,4624247Hz$

$\mu_(f)=\mu_R(f)\cdot f=0,041744 Hz $

$U(f)=k \mu_(f)=0,083488 Hz$

$f=(192,462±0,083) Hz$

## Calculo de la frecuencia con frecuencimetro  en modo frecuencimetro con gate 10s

Del punto C sabemos:

$f_x=\frac{10^M F_{BT}}{N}$

$f_x=\frac{10 10MHz}{519582}$

Ahora como estamos en modo frecuencia con gate de 10s sabemos que:

$f_x=\frac{N}{10s}$

Igualando y despejando:

$N=\frac{10s \cdot 10\cdot 10MHz}{519582}=1924,6240$

Como N es un contador descartamos los decimales y nos queda que:

N=1924

En este caso ahora 

$f_x=\frac{N}{10s}$

$f_x= 192,4 Hz$

El error de la estabilidad de la base de tiempo es el mismo, no tenemos error de disparo, y vamos a calcular el error de cuantización.

$\frac{\Delta N}{N}=\frac{1}{1924}=519,75 ppm$

$\frac{\Delta_f}{f}=±(\frac{\Delta f_{BT}}{f_{BT}}+\frac{\Delta N}{N})= 525,95 ppm$

$\mu(f)=\frac{ 525,95 ppm x 192,4 Hz}{sqrt3 }=0,058423 Hz$

$U(f)=K \cdot \mu(f)= 0,116847Hz$

f=(192,40±0,12) Hz

## Ejercicio 2

Se desea medir el tiempo de establecimiento de un generador de pulsos 

<img src="./Ej2.png" align="center" width="400" /> 

Punta de Prueba

* X1: C despreciable
* X10: CIN 5pF
* Tep : despreciable

Generador

* Tensión de salida 2Vpp 
* Impedancia de salida 50Ω 
* Trabaja sobre carga adaptada

Osciloscopio

* Ancho de Banda: 100MHz
* Impedancia de entrada: 1MΩ // 25pF
* Base de tiempo: 10ns/div. 
* Magnificador: X10
* Para la medición se utilizó la punta X1


* a) Determine el tiempo de establecimiento del generador.
* b) Si en vez de utilizar la punta X1, ahora se utiliza la punta X10, ¿cuál debería ser el tiempo de establecimiento indicado?
* c) Si el osciloscopio fuera de 500MHz de ancho de banda (utilizando la punta X1), ¿cuál debería ser el tiempo de establecimiento indicado?


$t_{ei}=\sqrt{t_{eg}^2+t_{eco}^2+t_{eo}^2}$

$t_{eg}=\sqrt{t_{ei}^2-t_{eco}^2-t_{eo}^2}$

$t_{ei}= 5 div \cdot 10 \frac{ns}{div})=50ns $

$t_{eo}=\frac{0,35}{BW}$

$t_{eo}=\frac{0,35}{100MHz}=3,5ns$

$t_{eco}=2,2 \tau$

Siendo $\tau=C_{osc} \cdot (Rt)$  

y siendo  Rt=1M$\Omega//50\Omega$

$t_{eco}=2,7498ns$

$t_{eg}=\sqrt{(50ns)^2-(2,7498ns)^2-(3,5ns)^2}=49,80149ns$

#### Medicion utilizando una punta x10 

En este caso ahora nos aparece

$t_{eco}=2,2 \tau$

Siendo $\tau=\frac{C_{osc}}{10}+5pf \cdot ((R_{osc}\cdot 10) //R_G)=0,87499ns$ 

$t_{eco}=1,92499ns$

$t_{eg}=\sqrt{(50ns)^2-(1,92499ns)^2-(3,5ns)^2}=49,8696ns$

#### Medicion utilizando un osc con un BW de 500MHz y una punta x1

$t_{eo}=\frac{0,35}{BW}$

$t_{eo}=\frac{0,35}{500MHz}=0,7ns$

$t_{eg}=\sqrt{(50ns)^2-(2,7498ns)^2-(0,7ns)^2}=49,91ns$


## Ejercicio 3

Se conecta un generador de pulsos a un DSO y luego de visualizar esta señal, se conecta a un contador universal

<img src="./Ej3.png" align="center" width="400" /> 

Especificaciones del DSO (Horizontal)

* Accuracy: ±(1 sample Interval + 100ppm x Reading + 0.6ns)
* Sample Interval = (s/div)/250 
* ∆Cursor =240.0ms
* BT =200ms


Especificaciones del contador universal

* Cantidad de digitos: 10
* FBT=10MHz.   
* Estabilidad 7.10-6 
* Envejecimiento 1.15-7 por mes (considerar 6 meses desde la calibración)
* SNR: 60dB


Deduzca el valor indicado en el display del contador universal con su INCERTIDUMBRE asociada a la medición de:

1. Periodo con el multiplicador en X1 y compare contra la incertidumbre del DSO. 
1. Periodo con el promediador en X1000.                                                               
1. Que indicaría el instrumento si se mide la frecuencia para tiempos de gate de 1s y 10s.   


Medición con el osciloscopio:

Ti=240ms

$\Delta T=(\frac{200ms}{250}+ 100ppm \cdot 240 ms + 0,6ns)= 824,0006 us $

$\mu_(T)=\frac{\Delta T}{\sqrt{3}}=475,7369 us$

Tomamos K=2 por adpotar la distribución normal

$U(T)=K \cdot \mu(f)= 951,4739 us$

$T=(240,0± 0,95) ms$


#### Midiendo periodo con el multiplicador x1


Tx=240ms

$N=Tx \cdot F_{BT}= 2400000$

$Ti=0,24 ms$

Estabilidad de la base de tiempo:

$\frac{\Delta f_{BT}}{f_{BT}}=7ppm+0,115\frac{ppm}{mes} \cdot 6 meses =7,69ppm$

Incertidumbre de cuantización:

$\frac{\Delta N}{N}=\frac{1}{2400000}=0,4166ppm$

$\frac{\Delta T}{T}=\frac{\Delta f_{BT}}{f_{BT}}+\frac{\Delta N}{N}=8,106ppm$

$\mu_(T)=\frac{\frac{\Delta T}{T}\cdot T}{\sqrt{3}}=1,1232 us$

Tomamos K=2 por adpotar la distribución normal

$U(T)=K \cdot \mu(f)= 2,246 us$

T=(240000,0 ±2,2)us

#### Midiendo periodo con el promediador x1000

Tx=240ms

$N=1000 \cdot  Tx \cdot F_{BT}= 2400000000$

$Ti=240,0000 us$


Estabilidad de la base de tiempo:

$\frac{\Delta f_{BT}}{f_{BT}}=7ppm+0,115\frac{ppm}{mes} \cdot 6 meses =7,69ppm$

Incertidumbre de cuantización:

$\frac{\Delta N}{N}=\frac{1}{2400000000}=0,0004166ppm$

$\frac{\Delta T}{T}=\frac{\Delta f_{BT}}{f_{BT}}+\frac{\Delta N}{N}=8,106ppm$

$\mu_(T)=\frac{\frac{\Delta T}{T}\cdot T}{\sqrt{3}}=1,06561 us$

Tomamos K=2 por adpotar la distribución normal

$U(T)=K \cdot \mu(f)= 2,1312 us$

T=(240000,0 ±2,1)us


#### Indicación si se mide frecuencia para gate de 1 y 10 s

#### Si gate=1s

$N=\frac{1}{240ms}=4$

Indicación del instrumento:

fi=4Hz

Estabilidad de la base de tiempo:

$\frac{\Delta f_{BT}}{f_{BT}}=7ppm+0,115\frac{ppm}{mes} \cdot 6 meses =7,69ppm$

Incertidumbre de cuantización:

$\frac{\Delta N}{N}=\frac{1}{4}=250000ppm$

$\frac{\Delta f}{f}=\frac{\Delta f_{BT}}{f_{BT}}+\frac{\Delta N}{N}=250007,69ppm$

$\mu_(f)=\frac{\frac{\Delta f}{f}\cdot f}{\sqrt{3}}=0,5773 Hz$

Tomamos K=2 por adpotar la distribución normal

$U(f)=K \cdot \mu(f)= 1,1547 Hz$

f=(4,0±1,2)Hz


#### Si gate=10s

$N=\frac{10}{240ms}=41$

Indicación del instrumento:

fi=4,1Hz

Estabilidad de la base de tiempo:

$\frac{\Delta f_{BT}}{f_{BT}}=7ppm+0,115\frac{ppm}{mes} \cdot 6 meses =7,69ppm$

Incertidumbre de cuantización:

$\frac{\Delta N}{N}=\frac{1}{4}=24390,2439ppm$

$\frac{\Delta f}{f}=\frac{\Delta f_{BT}}{f_{BT}}+\frac{\Delta N}{N}=24397,93ppm$

$\mu_(f)=\frac{\frac{\Delta f}{f}\cdot f}{\sqrt{3}}=0,05773 Hz$

Tomamos K=2 por adpotar la distribución normal

$U(f)=K \cdot \mu(f)= 0,1155 Hz$

f=(4,10±0,12)Hz


En los puntos Ay B no tuvimos en cuenta el error de disparo, pese a tener el dato de la relación señal a ruido, porque no sabemos la pendiente de la función, y si la consideramos ideal, su dereviada daría infinita, por tanto el error de disparo 0. Tampoco sabemos el valor de la tensión a la cual se dispara. En caso que este valor no fuera infinito, y comparativamente tendriamos menos error de disparo en el caso del promediado por 1000, dado que este valor disminuye el error de disparo. 

En consecuencia vemos que para este caso, el modo promediado es la mejor opción. Queda en evidencia que medir frecuencia, para valores bajos de frecuencia, hacen que el error de cuantización pese mucho en el calculo total de la incertidumbre. 

Comparativamente vemos que tenemos mucha menor incertidumbre utilizando un contador de frecuencia en modo periodo que utilizando el osciloscopio. 

## Ejercicio 4

Se desea comprobar la frecuencia de trabajo de un driver de tira led RGB, con valor nominal de 800KHz según hoja de datos.

Se dispone de un contador universal de 8 dígitos con Fbt=1MHz ± 1ppm, tiempos de compuerta de 0.1s, 1s y 10s, promediado de 1, 10 y 100 ciclos.

Se realiza un serie de 5 mediciones, obteniéndose en display las siguientes indicaciones:

|Medición|
| -- | 
| 00800.013 KHz |
| 00800.011 KHz |
| 00800.014 KHz |
| 00800.011 KHz |
| 00800.010 KHz |

Responda:
* ¿Cuál fue el tiempo de compuerta seteado?
* Calcule incertidumbre combinada de la frecuencia medida
* Proponga un cambio de seteo que permita mejorar la incertidumbre tipo B calculada en el punto anterior. Recalcule  incertidumbre tipo B y compare.




Tiempo de compuerta seteado:

$TC=\frac{800013}{800,013KHz}=1s$

Calculo de la incertidumbre:

Tipo A:


In [14]:
import numpy as np
import statistics as st


Freqs=np.array([800013,800011,800014,800011,800010])
Freqsm=np.mean(Freqs)
SFreqs=st.stdev(Freqs)
uaFreqs=SFreqs/np.sqrt(len(Freqs))
print(Freqsm)
print(SFreqs)
print(uaFreqs)

800011.8
1.4142135623730951
0.6324555320336759


Tipo B:

Estabilidad de la base de tiempo:

$\frac{\Delta f_{BT}}{f_{BT}}=1ppm$

Incertidumbre de cuantización:

$\frac{\Delta N}{N}=\frac{1}{800011}=1,24998 ppm$

$\frac{\Delta_f}{f}=±(\frac{\Delta f_{BT}}{f_{BT}}+\frac{\Delta N}{N})= 2,24998 ppm$

$\mu_B(f)=\frac{ 2,24998 ppm x 800011.8 Hz}{\sqrt3 }=1,032923 Hz$

$\mu_c=\sqrt{\mu_A(f)^2+\mu_B(f)^2}=1,21116882553$

$\frac{\mu_B(f)}{\mu_A(f)}=1,633$

No podemos asumir que ninguna de las incertidumbres sea dominante por lo tanto asumimos un K=2

$U(f)=K \cdot \mu(f)= 2,42Hz$

f=(800011,8±2,4) Hz

# Cambio de seteo para minimizar la incertidumbre

Para mejorar la incertidumbre lo mejor es buscar un tiempo de compuerta que nos haga llegar al contador más alto, para que la incertidumbre de cuantización sea menor.No es conveniente cambiar a modo periodo dado que nos encontramos en frecuencias altas, cercanas al MHz

Si tomamos un tiempo de compuerta de 10s

$10s \cdot 800,013KHz=8000130$

Es decir agregamos un digito más a nuestra visualización, a modo de comparación tomamos la misma incertidumbre tipo A, y que el valor medio indicado es el mismo con un digito más

Incertidumbre de cuantización:

$\frac{\Delta N}{N}=\frac{1}{8000118}=0,124998 ppm$

$\frac{\Delta_f}{f}=±(\frac{\Delta f_{BT}}{f_{BT}}+\frac{\Delta N}{N})= 1,124998 ppm$

$\mu_B(f)=\frac{ 1,124998 ppm x 800011.8 Hz}{\sqrt3}=0,519621 Hz$

$\mu_c=\sqrt{\mu_A(f)^2+\mu_B(f)^2}=0,8184966$

$\frac{\mu_B(f)}{\mu_A(f)}=0,8247$

No podemos asumir que ninguna de las incertidumbres sea dominante por lo tanto asumimos un K=2

$U(f)=K \cdot \mu(f)= 1,63699Hz$

f=(800011,8±1,6) Hz



